ui/src/app/components/gke-app-configuration-panels/create-gke-app-button.tsx

```diff
@@ -75,6 +86,13 @@ export function CreateGkeAppButton({
           appMinDiskSize[createAppRequest.appType]
         } GB`,
     ],
+    [
+      wouldDecreaseDiskSize,
+      () =>
+        `The specified disk size (${createAppRequest.persistentDiskRequest.size} GB) is smaller than ` +
+        `the existing unattached persistent disk size of ${existingDisk.size} GB. ` +
+        'Preventing creation because this would cause data loss.',
```

Comment:
We allow users to do this for Runtimes, after making them confirm that data loss is OK.  But that logic is complex and not on our roadmap for GKE Apps.

An equivalent action users can take here is to delete the disk, and then create the GKE App with a new disk of any size.